### PR TITLE
Provide error message if process file is unavailable due to permissions for teleport start

### PR DESCRIPTION
**Pull request by stevenGravy**
Provides information when unable to access sqllite.db path.  Due to load order this stopped the effectiveness of the messages
on permissions done in #10044. 

current error
```bash
2023-04-20T14:17:14-04:00             INFO Starting Teleport v14.0.0-dev with a config file located at "teleport.yaml" common/teleport.go:548
ERROR: initialization failed
mkdir data/proc: permission denied
```

to

```bash
2023-04-20T14:16:22-04:00             INFO Starting Teleport v14.0.0-dev with a config file located at "teleport.yaml" common/teleport.go:548
ERROR: Teleport does not have permission to write to /var/lib/teleport/proc/sqlite.db
initialization failed
mkdir data/proc: permission denied
```

### CODE REVIEW

File: lib/backend/lite/lite.go [hunk 1]
```diff
@@ -205,6 +206,9 @@ func NewWithConfig(ctx context.Context, cfg Config) (*Backend, error) {
 	// Ensure that the path to the root directory exists.
 	err := os.MkdirAll(cfg.Path, os.ModeDir|defaultDirMode)
 	if err != nil {
+		if errors.Is(err, fs.ErrPermission) {
+			return nil, trace.Wrap(err, "Teleport does not have permission to write to: %v. Ensure that you are running as a user with appropriate permissions.", path)
```
Comment (zmb3) [inline lib/backend/lite/lite.go:210]:
```suggestion
			return nil, trace.AccessDenied("Teleport does not have permission to write to %v", path)
```

What does the error look like if you do this instead?

Comment (stevenGravy) [reply lib/backend/lite/lite.go:210]: Here you go

```
INFO             Starting Teleport v14.0.0-dev common/teleport.go:546
ERROR: initialization failed
Teleport does not have permission to write to /var/lib/teleport/proc/sqlite.db
```

Comment (zmb3) [reply lib/backend/lite/lite.go:210]: Let's do that.

Comment (stevenGravy) [reply lib/backend/lite/lite.go]: updated @zmb3

**Review by zmb3 — APPROVED** [review]

**Review by mdwn — APPROVED** [review]

**Pull request merged**